src/main/java/teammates/ui/webapi/UpdateNotificationAction.java

```diff
@@ -34,7 +35,7 @@ public JsonResult execute() throws InvalidHttpRequestBodyException {
                 .build();
 
         try {
-            return new JsonResult(new NotificationData(logic.updateNotification(newNotification)));
+            return new JsonResult(new NotificationData(sqlLogic.updateNotification(newNotification)));
```

Comment:
Can we fetch the notification then use the setters to update rather than create a detached object. This ensures that it's in a managed state (less chance of random errors when session is flushed, slight performance benefits as well).